crates/p2p/src/swarm/errors.rs

```diff
@@ -38,9 +38,13 @@ pub enum ProtocolError {
     #[error("Failed to listen: {0}")]
     Listen(#[from] TransportError<io::Error>),
 
-    /// The event channel somehow is closed.
-    #[error("Events channel closed: {0}")]
-    EventsChannelClosed(Box<dyn std::error::Error + Sync + Send>),
+    /// The gossip channel somehow is closed.
+    #[error("Gossip channel closed: {0}")]
+    GossipEventsChannelClosed(Box<dyn std::error::Error + Sync + Send>),
```

Comment:
Can you add a top level import of `std::error` and then change this to?

```suggestion
    GossipEventsChannelClosed(Box<dyn error::Error + Sync + Send>),
```

This makes it easier to see what imports this module, aka file, uses by checking the top-level imports.